src/lib/apputils/net-server.c

```diff
@@ -753,7 +764,7 @@ setup_socket(struct socksetup *data, struct bind_address *ba,
     }
 
     /* Try to turn on pktinfo for UDP wildcard sockets. */
-    if (ba->type == UDP && ba->address == NULL) {
+    if (ba->type == UDP && is_wildcard(sock_address)) {
         krb5_klog_syslog(LOG_DEBUG, _("Setting pktinfo on socket %s"),
```

Comment:
I think the following conditions might have issues: Point-to-Point, Loopback, and Multicast.

I do think if pktinfo fails, the code should use the default socket address (whether or not it does currently is something that should be examined)... I agree pktinfo may not always work but I think it presumptuous to assume "wildcard" is the only condition where it is required. I would be curious what you find.